nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-administration/src/main/java/org/apache/nifi/admin/service/impl/StandardKeyService.java

```diff
@@ -118,11 +118,16 @@ public void deleteKey(String identity) {
             transaction = transactionBuilder.start();
 
             // delete the keys
-            DeleteKeysAction deleteKeys = new DeleteKeysAction(identity);
-            transaction.execute(deleteKeys);
-
-            // commit the transaction
-            transaction.commit();
+            DeleteKeyAction deleteKey = new DeleteKeyAction(keyId);
+            Integer rowsDeleted = transaction.execute(deleteKey);
+
+            // commit the transaction if we found one and only one matching keyId/user identity
+            if (rowsDeleted == 1) {
```

Comment:
@mcgilman 

This method previously deleted by 'identity', which could potentially delete multiple rows. I've changed it to delete by row/key ID. Could there be side effects for this?

Reply:
I believe that the identity is marked as unique [1]. What condition resulted in deleting multiple rows?

[1] https://github.com/apache/nifi/blob/main/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-administration/src/main/java/org/apache/nifi/admin/KeyDataSourceFactoryBean.java#L47

Reply:
I don't believe there was a specific condition I knew of, I just wasn't sure. I think the naming and possibly a comment had implied to me it was possible to delete multiple rows using the identity. Sounds like that's not an issue. As such, the method can remain as shown.